### PR TITLE
[Merged by Bors] - Refactor `Camera` methods and add viewport rect 

### PR DESCRIPTION
**Pull request by aevyrie**
While working on a refactor of `bevy_mod_picking` to include viewport-awareness, I found myself writing these functions to test if a cursor coordinate was inside the camera's rendered area.

# Objective

- Simplify conversion from physical to logical pixels
- Add methods that returns the dimensions of the viewport as a min-max rect

---

## Changelog

- Added `Camera::to_logical`
- Added `Camera::physical_viewport_rect`
- Added `Camera::logical_viewport_rect`

### CODE REVIEW

**Comment by Weibye** [comment]
> * 🚲 🏠 opportunity: should `Camera::physical_to_logical` be renamed to `Camera::to_logical` to better follow the [rust api guidelines](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv), or would that just obscure the `from` side of the conversion?

Isn't `T::from()` usually meaning that you create a new object of that type? In which case `Camera::to_logical(input: physical)` should be enough to convey the full meaning given that the input parameter is usefully named / has docs.

> * Bevy has some `Rect` types. [This](https://dev-docs.bevyengine.org/bevy/sprite/struct.Rect.html) has the same data structure, but lives in bevy_sprite. [This](https://dev-docs.bevyengine.org/bevy/prelude/struct.UiRect.html) `Rect` lives in bevy_ui, but it uses a different structure (left/right/top/bottom).

^ I really would like to get the geometric-shapes RFC merged so that we could start building and reusing for these purposes

File: crates/bevy_render/src/camera/camera.rs [hunk 1]
```diff
@@ -104,21 +104,45 @@ impl Default for Camera {
 }
 
 impl Camera {
+    /// Converts a physical size in this `Camera` to a logical size.
+    #[inline]
+    pub fn physical_to_logical(&self, input: UVec2) -> Option<Vec2> {
+        let scale = self.computed.target_info.as_ref()?.scale_factor;
+        Some((input.as_dvec2() / scale).as_vec2())
+    }
```
Comment (Weibye) [inline crates/bevy_render/src/camera/camera.rs:112]: I think you suggestion of naming this `to_logical` is good if we rename the input parameter as well.

```suggestion
    pub fn to_logical(&self, physical_size: UVec2) -> Option<Vec2> {
        let scale = self.computed.target_info.as_ref()?.scale_factor;
        Some((input.as_dvec2() / scale).as_vec2())
    }
```

**Review by Weibye — APPROVED** [review]
This looks good to me :)

When it comes to `Physical` vs `Logical`, we could look into creating or reusing a type for it similar to what `winit` is doing: https://docs.rs/winit/0.21.0/winit/dpi/enum.Position.html

**Comment by aevyrie** [comment]
> ^ I really would like to get the geometric-shapes RFC merged so that we could start building and reusing for these purposes

Preaching to the choir. :slightly_smiling_face: 

> When it comes to Physical vs Logical, we could look into creating or reusing a type for it similar to what winit is doing: https://docs.rs/winit/0.21.0/winit/dpi/enum.Position.html

I like this, though we may want to provide some auto deref impls to reduce any added overhead.

**Review by alice-i-cecile — APPROVED** [review]
bors r+

**Pull request closed without merging**